tests/envs/test_envs.py

```diff
@@ -120,3 +128,20 @@ def test_render_modes(spec):
             new_env.render()
             new_env.close()
     env.close()
+
+
+@pytest.mark.parametrize(
+    "env",
+    all_testing_initialised_envs,
+    ids=[env.spec.id for env in all_testing_initialised_envs],
+)
+def test_pickle_env(env: gym.Env):
+    env.np_random, _ = seeding.np_random(0)
```

Comment:
You're not resetting the env before pickling which is likely to mess things up (and I don't think it's supported behavior anyways, since you need to reset an env before doing literally anything with it)

Reply:
I believe that pickling is used by RLLib vectorisation code so they make environments then pickle for making multiple copies.